### PR TITLE
In setup.py, inline the packages that need to be installed into setup().

### PR DESCRIPTION
**Pull request by anntzer**
... instead of hiding that behind a function call in setupext.

(See also #13542 which does the same with {install,setup}_requires.)

Also always install the test source files -- setting `tests = False` in
setup.cfg now only ignores the test data (this is also consistent with
the contents of `matplotlib/tests/__init__.py which is clearly intended
to handle the case where the test source files are installed but not the
baseline images).  This is done for simplicity of implementation and
also to make it possibly easier to separately install the baseline
images (as these are now pure "data") in the future (WIP in  #11732).

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -275,12 +271,12 @@ def run(self):
         interfaces and hardcopy output formats.
         """,
         license="PSF",
-        packages=packages,
-        namespace_packages=namespace_packages,
-        platforms='any',
-        py_modules=py_modules,
+        platforms="any",
+        package_dir={"": "lib"},
+        packages=find_packages("lib"),
```
Comment (tacaswell) [inline setup.py]: Does this always install the test modules now?

**Comment by tacaswell** [comment]
Deleted my comment because I failed to read before I typed.

Given the discussion we had with the Arch packagers in https://github.com/matplotlib/matplotlib/issues/13993 is this going to cause the same problem?

Having clean data-only wheels for the test data does seem like a good carrot.

**Comment by anntzer** [comment]
We can ping @ArchangeGabriel here to ask :)  This time it is definitely intentional to always install the test source files, just leaving the baseline images optional; I think if downstream packagers want to also remove the tests they can just yank them out a posteriori.

**Comment by ArchangeGabriel** [comment]
It’s fine, we can indeed just remove the test files afterwards if they are not spread over hundreds of directories. ;)

Also since we don’t pass the baseline images comparisons (because of different freetype), this change could actually leads to simplification of our building scripts if testing skip baseline images tests when they are missing.

**Comment by anntzer** [comment]
You can just `rm **/tests/`, I think.  Not sure what the current behavior is on missing baseline images, but I guess we can make them just ignored in that case, indeed.

**Comment by ArchangeGabriel** [comment]
Ignoring the image tests would be awesome because it means we would actually be able to automatically and easily verify no other tests failed. :)

File: setup.py [hunk 2]
```diff
@@ -236,9 +235,6 @@ def run(self):
         # Now collect all of the information we need to build all of the
         # packages.
         for package in good_packages:
-            packages.extend(package.get_packages())
-            namespace_packages.extend(package.get_namespace_packages())
-            py_modules.extend(package.get_py_modules())
```
Comment (QuLogic) [inline setup.py]: These variables can be removed now, no?

Comment (anntzer) [reply setup.py]: indeed, done.

**Comment by ArchangeGabriel** [comment]
I think that [this part of setup.cfg](https://github.com/matplotlib/matplotlib/blob/v3.2.x/setup.cfg.template#L13) should have been updated to say that what is controlled here is only about data, not actually packages (tests being installed anyway now). What is your opinion on this?

**Comment by anntzer** [comment]
Sounds fair, do you want to open a PR for that?

**Comment by ArchangeGabriel** [comment]
Also, currently having `tests = False` outputs `tests: no  [skipping due to configuration]` but that does not actually forbid to run tests. So yes I might open a PR. Should I target master first and then backport to 3.2.x?

**Comment by anntzer** [comment]
yes

**Comment by ArchangeGabriel** [comment]
OK, made some attempt at https://github.com/matplotlib/matplotlib/pull/15480.

**Pull request merged**